From ff6b00c5560c39272cef1804001903eda604b05a Mon Sep 17 00:00:00 2001
From: Praveen Paneri <praveen.paneri@intel.com>
Date: Tue, 12 Jan 2016 17:41:21 +0530
Subject: [PATCH 5/6] SQUASHME! [VPG]: drm/i915: Do NULL pointer check before
 dereferencing it

tmpreq->ctx is checked for NULL after it is dereferenced.
Fixing this by moving the NULL check before dereferencing

SQUASHME! - This patch should be squashed into the below patch.

Subject: FOR_UPSTREAM [VPG]: drm/i915: Added gen8 support for
 Timeout Detection Recovery for engine hangs
Author: Tomas Elf <tomas.elf@intel.com>
Change-Id: I45de965b4a81d93c135a74248eb22a8a5c816e9f

Change-Id: I905bbcd533ab242817f025c94f1e0c26a64b443e
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-17848
Signed-off-by: Praveen Paneri <praveen.paneri@intel.com>
Reviewed-on: https://android.intel.com:443/463665
---
 drivers/gpu/drm/i915/intel_lrc.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_lrc.c b/drivers/gpu/drm/i915/intel_lrc.c
index c408ea6..a642cf9 100644
--- a/drivers/gpu/drm/i915/intel_lrc.c
+++ b/drivers/gpu/drm/i915/intel_lrc.c
@@ -927,7 +927,7 @@ intel_execlists_TDR_get_submitted_context(struct intel_engine_cs *ring,
 	tmpreq = list_first_entry_or_null(&ring->execlist_queue,
 		struct intel_ctx_submit_request, execlist_link);
 
-	if (tmpreq) {
+	if (tmpreq && tmpreq->ctx) {
 		sw_context =
 			intel_execlists_ctx_id((tmpreq->ctx)->engine[ring->id].state);
 
@@ -945,8 +945,7 @@ intel_execlists_TDR_get_submitted_context(struct intel_engine_cs *ring,
 			 * assert if we do). Just rely on the execlist code to provide
 			 * indirect protection.
 			 */
-			if (tmpreq->ctx)
-				tmpctx = tmpreq->ctx;
+			tmpctx = tmpreq->ctx;
 
 			if (ctx)
 				i915_gem_context_reference(tmpctx);
-- 
1.9.1

